java-checks-test-sources/default/src/main/java/checks/MathClampMethodsCheckSample.java

```diff
@@ -0,0 +1,98 @@
+package checks;
+
+public class MathClampMethodsCheckSample {
+
+  public void testNoncompliant(int min, int max, int value) {
+    int clampedValueConditional1 = value > max ? max : value < min ? min : value; // Noncompliant [[sc=36;ec=81]] {{Use "Math.clamp" instead of a conditional expression.}}
```

Comment:
The comparison logic to analyze `value > max  ? max : (value < min ? min : value)` does not support some equivalent inversions like `value > max` <=> `max < value`. So we have those false-negatives:
```
    var testCases = new int[] {
      value > max ? max : (value < min ? min : value), // Noncompliant
      max < value ? max : (value < min ? min : value), // false-negative
      value <= max ? (value < min ? min : value) : max, // false-negative
      max >= value ? (value < min ? min : value) : max, // false-negative
      max >= value ? (min > value ? min : value) : max, // false-negative
      max >= value ? (value >= min ? value : min) : max, // false-negative
      value > max ? max : (min > value ? min : value), // Noncompliant
      value > max ? max : (value >= min ? value : min), // Noncompliant
      value > max ? max : (min <= value ? value : min) // Noncompliant
    };
```